app/models/merchant.rb

```diff
@@ -35,7 +35,31 @@ def self.where_by_params(param)
   end
 
   def self.random
-    all.shuffle.first
+    order('RANDOM()').first
+  end
+
+  def total_revenue
+    invoices
+    .joins(:transactions, :invoice_items)
+    .merge(Transaction.successful)
+    .sum("invoice_items.quantity * invoice_items.unit_price")
+  end
+
+  def total_revenue_by_date(date)
```

Comment:
this method is _almost_ identical to your `total_revenue` method. could we DRY up this code somehow?